### PR TITLE
use XHP for hhast-inspect

### LINKED ISSUE #323: HHAST-Inspect should use XHP
Currently doesn't as the code is namespaced, and XHP was unusable in namespaces until recently.

### PR DESCRIPTION
**Pull request by fredemmott**
This is mostly trivial, with a tiny bit of refactoring. Found a small
bug - pre/code open and close order were mismatched.

This should probably be split out to a separate repo; having HHAST
depend on xhp-lib isn't really acceptable, but neither is using HTML
strings when there's a better alternative.

fixes #323

### CODE REVIEW

File: src/__Private/Inspector/InspectorCLI.hack
Comment (fredemmott) [inline src/__Private/Inspector/InspectorCLI.hack:19]: Hackfmt bug I guess :)

File: src/__Private/Inspector/InspectorCLI.hack
Comment (fredemmott) [inline src/__Private/Inspector/InspectorCLI.hack:18]: @lexidor previously suggested a whitelist for UseStatementWithAs - this should be configurable, but we potentially want to whitelist Facebook\XHP by default

**Review by jjergus — APPROVED** [review]

**Review by lexidor — APPROVED** [review]
Code changes are indeed mostly trivial.
How about that circular dependency?
Should the inspect functionality be its own _small_ package?

**Comment by lexidor** [comment]
Test failure is strange.
Testing on ubuntu against 4.72 resulted in the following message:
```
Get:30 https://dl.hhvm.com/ubuntu bionic-4.72/main amd64 Packages [1272 B]
Fetched 20.6 kB in 3s (8004 B/s)
+ sudo apt-get install -y hhvm
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
hhvm is already the newest version (4.75.0-1~bionic).
0 upgraded, 0 newly installed, 0 to remove and 27 not upgraded.
```

Mac fails (correctly), since xhp 4 can't run on hhvm 4.72.

**Comment by fredemmott** [comment]
Closing: hhast-inspect is being split to a new repository for this because of the circular dependency. Basically ready to go, waiting for final approval on the new repo

**Pull request closed without merging**
